src/main/main.ts

```diff
@@ -188,214 +189,195 @@ function initializeApp() {
-        console.log(
-          "Start to extract the tarball archive",
-          dlPath,
-          "to",
-          extractPath
-        );
-        try {
-          await spawnPromise(
-            "tar",
-            [`xvf${bz2 ? "j" : "z"}`, dlPath, "-C", extractPath],
-            { capture: ["stdout", "stderr"] }
-          );
-        } catch (e) {
-          console.error(`${e}:\n`, e.stderr);
-          throw e;
-        }
-        console.log(
-          "The tarball archive",
-          dlPath,
-          "has extracted to ",
-          extractPath
-        );
-      } else {
-        console.warn("Not supported platform.");
-        return;
-      }
+  if (win == null) return;
 
-      // 압축을 푼 뒤 압축 파일은 제거합니다.
-      await fs.promises.unlink(dlPath);
+  console.log("peerVersionExtra (hex):", peerVersionExtra);
+  const buffer = Buffer.from(peerVersionExtra, "hex");
+  console.log("peerVersionExtra (bytes):", buffer);
+  const extra = decode(buffer) as BencodexDict;
+  console.log("peerVersionExtra (decoded):", JSON.stringify(extra)); // 다른 프로세스라 잘 안보여서 JSON으로...
+  const macOSBinaryUrl = extra.get("macOSBinaryUrl") as string;
+  const windowsBinaryUrl = extra.get("WindowsBinaryUrl") as string;
+  console.log("macOSBinaryUrl: ", macOSBinaryUrl);
+  console.log("WindowsBinaryUrl: ", windowsBinaryUrl);
+  const downloadUrl =
+    process.platform === "win32"
+      ? windowsBinaryUrl
+      : process.platform === "darwin"
+      ? macOSBinaryUrl
+      : null;
 
-      // 설정 합치기
-      const newConfig = JSON.parse(
-        await fs.promises.readFile(configPath, { encoding: "utf-8" })
-      );
-      const config = {
-        ...bakConfig,
-        ...newConfig,
-      };
-      await fs.promises.writeFile(configPath, JSON.stringify(config), "utf-8");
+  if (downloadUrl == null) return; // 지원 안 하는 플랫폼이니 무시.
+
+  // TODO: 이어받기 되면 좋을 듯
+  const options: ElectronDLOptions = {
+    onStarted: (downloadItem: DownloadItem) => {
+      console.log("Starts to download:", downloadItem);
+    },
+    onProgress: (status: IDownloadProgress) => {
+      const percent = (status.percent * 100) | 0;
       console.log(
-        "The existing and new configuration files has been merged:",
-        config
+        `Downloading ${downloadUrl}: ${status.transferredBytes}/${status.totalBytes} (${percent}%)`
       );
+      win?.webContents.send("update download progress", status);
+    },
+    directory: app.getPath("temp"),
+  };
+  console.log("Starts to download:", downloadUrl);
+  let dl: DownloadItem | null | undefined;
+  try {
+    dl = await download(win, downloadUrl, options);
+  } catch (error) {
+    win?.webContents.send("go to error page", "download-binary-failed");
+    throw new DownloadBinaryFailedError(downloadUrl);
+  }
 
-      lockfile.unlockSync(lockfilePath);
-      console.log(
-        "Removed 'encounter different version' lockfile at ",
-        lockfilePath
+  win?.webContents.send("update download complete");
```

Comment:
No need to check null.
```suggestion
  win.webContents.send("update download complete");
```